### ISSUE TITLE
Machine readable date in JSON output?

### ISSUE THREAD

**Issue by thomthom** [google/benchmark#825]
I was looking at parsing the JSON output from the JSON formatter. But it appear that the `date` field in `context` is formatted to the locale of the current user: https://github.com/google/benchmark/blob/master/src/json_reporter.cc#L109-L110

I'd argue that in a machine readable format such as JSON it would be better to use [ISO 8601](https://www.iso.org/iso-8601-date-and-time-format.html) to ensure the date can be read without concerns of having to know the locale and timezone of the origin of the JSON file.

Would you accept a PR with such a change?

**Comment by LebedevRI** [comment]
That does look like a bug to me.

**Comment by thomthom** [comment]
hm... Just realized that `DateTimeString` uses "%F %T" ISO 8601 formatting on all platforms other than Windows. (Not sure what "%x %X" means, I don't see it in the cppreference.) But it doesn't output ISO 8601 on my Windows machine.

**Comment by thomthom** [comment]
So there are two issues here - it uses local time for the JSON date instead of UTC. And Windows has an non-standard formatting that deviates from ISO 8601.

**Comment by LebedevRI** [comment]
> it uses local time for the JSON date instead of UTC.

That actually sounds correct, assuming the time zone is specified in the output string.

**Comment by thomthom** [comment]
> That actually sounds correct, assuming the time zone is specified in the output string.

It isn't. (Not with 1.5.0 at least) The string isn't fully ISO 8601 compliant - no timezone indication.

Here is an example from my machine, using regional format for "English (United Kingdom)". But I'm in Norway so I'm UTC+1. And it's currently summer time so I'm currently UTF+2

```json
{
  "context": {
    "date": "06/02/19 00:51:37",
    "host_name": "HIVE-MIND",
    "executable": "SUbD_Benchmark.exe",
    "num_cpus": 16,
    "mhz_per_cpu": 2994,
    "cpu_scaling_enabled": false,
    "caches": [
      {
        "type": "Data",
        "level": 1,
        "size": 32768000,
        "num_sharing": 2
      },
      {
        "type": "Instruction",
        "level": 1,
        "size": 65536000,
        "num_sharing": 2
      },
      {
        "type": "Unified",
        "level": 2,
        "size": 524288000,
        "num_sharing": 2
      },
      {
        "type": "Unified",
        "level": 3,
        "size": 8388608000,
        "num_sharing": 8
      }
    ],
    "load_avg": [],
    "library_build_type": "release"
  },
  "benchmarks": []
```

**Comment by thomthom** [comment]
To provide a bit more context, I was going to push the JSON output from Google Benchmark to a server - and the JSON data is currently not enough for me to reliable parse the date from the server as it doesn't know the locale and timezone of the clients..

**Comment by thomthom** [comment]
Hm... it doesn't even match the date format in my system settings...

![image](https://user-images.githubusercontent.com/192418/58761422-907ea500-8544-11e9-96da-dc971cb76440.png)

I'm seeing MM/DD/YYYY in the JSON, but my regional settings days DD/MM/YYYY

**Comment by dominichamon** [comment]
It's likely little thought has been put into the time output. If there's
something better/consistent we could be doing (ISO + UTC or ISO + TZ both
sound reasonable) then I'm totally open to it.

Dominic Hamon | Google
*There are no bad ideas; only good ideas that go horribly wrong.*

**Comment by thomthom** [comment]
ISO 8601 date and time formatting will take care of UTC and time zones. Made to be reliably parsed and exchanged.

**Comment by brianwolfe** [comment]
@thomthom @dominichamon I think this issue is fixed with the change linked above.

**Comment by LebedevRI** [comment]
Fixed by #965.

**Comment by thomthom** [comment]
Great! Thanks for the heads up - now I can remove the timestamp parsing kludge on my end. :)
